src/extensions/default/analyzers/Microsoft.DotNet.UpgradeAssistant.Extensions.Default.Analyzers.Common/GeneralInvocationExpression.cs

```diff
@@ -0,0 +1,57 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
+using System.Linq;
+using Microsoft.CodeAnalysis;
+using CS = Microsoft.CodeAnalysis.CSharp;
+using VB = Microsoft.CodeAnalysis.VisualBasic;
+
+namespace Microsoft.DotNet.UpgradeAssistant.Extensions.Default.Analyzers.Common
+{
+    public class GeneralInvocationExpression
+    {
+        private readonly SyntaxNode _invocationExpression;
+
+        private GeneralInvocationExpression(SyntaxNode syntaxNode)
+        {
+            // the only valid way to construct one is to try parse
+            // this enables validation to assert we're providing a wrapper around a valid SyntaxNode
+            _invocationExpression = syntaxNode ?? throw new ArgumentNullException(nameof(syntaxNode));
```

Comment:
Why not just use IsInvocationExpression to do the necessary validation in the ctor and make it public? That seems like it would be more direct than having the TryParse helper (which isn't really *parsing* anything, per se).

Reply:
checks to see if a node is of the correct type before we allow constructing it. because the two are paired together, it seemed like the tryParse flow.

maybe another option is something like
```
if (!syntaxNode.IsInvocationExpression())
{
   return;
}
var generalExpression = syntaxNode.AsGeneralInvocationExpression();
```

Reply:
I went with "General" since "Generic" in csharp has a significant meaning. Does it seem appropriate? Feels a bit off to me but I haven't come up with a better name yet. Maybe `ProxyInvocationExpression`
@mjrousos 
@twsouthwick 
@sunandabalu

Reply:
General seems ok to me.